mujoco_py/tests/test_cymj.py

```diff
@@ -651,6 +650,33 @@ def test_sensors():
 
 
 @requires_rendering
+def test_lock():
+    # Tests whetever multithreaded rendering is safe.
+    model = load_model_from_xml(BASIC_MODEL_XML)
+    sim = MjSim(model)
+    sim.render(100, 100)
+    def func():
+        nonlocal sim
+        sim.data.qpos[:] = 0.0
+        sim.forward()
+        img1 = sim.render(width=40, height=40, camera_name="camera1")
```

Comment:
You could use a shared `threading.Event` variable here: let all the threads wait for it, then set it to make sure more of the threads run concurrently. In the current implementation, there's always the chance that one thread finishes before the next one starts. Using an event reduces (but doesn't eliminate) it.

If you do this, you could also also do `thread.start()` right after instantiation.